### ISSUE TITLE
XLA `argmax`/`argmin` returns wrong index when NaN is present for non-power-of-2 sizes

### ISSUE THREAD

**Issue by wuyii8941** [tensorflow/tensorflow#118647]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

 TensorFlow 2.22.0-dev20260429

### Custom code

Yes

## Summary

Under XLA (`jit_compile=True`), `tf.math.argmax` and `tf.math.argmin` return incorrect indices when the input contains NaN and the array size is not a power of 2 (i.e., sizes 3, 5, 6, 7, 9, 10, ...). Instead of returning the index of the maximum/minimum non-NaN element, XLA returns the **last index** of the array.

Power-of-2 sizes (2, 4, 8, 16, ...) are unaffected.

Eager mode always returns the correct index.

## Scope

**Affected sizes**: Every non-power-of-2 size from 3 to 19 tested. The pattern is consistent.

| Array size | Power of 2? | argmax(all-NaN) eager | argmax(all-NaN) XLA | Correct? |
|---|---|---|---|---|
| 2 | yes | 0 | 0 | OK |
| 3 | no | 0 | **2** | BUG |
| 4 | yes | 0 | 0 | OK |
| 5 | no | 0 | **4** | BUG |
| 8 | yes | 0 | 0 | OK |
| 10 | no | 0 | **9** | BUG |
| 16 | yes | 0 | 0 | OK |
| 17 | no | 0 | **16** | BUG |

**Affects both argmax and argmin**:

```python
x = tf.constant([1.0, float('nan'), 3.0])

# argmin: eager=0 (1.0 is min), XLA=2 (wrong)
print(tf.function(tf.math.argmin, jit_compile=True)(x).numpy())  # 2
```

**NaN position matters**: The bug triggers when NaN is at any position that is not before the actual extremum in the XLA reduction tree. For size-3 arrays:

| Input | argmax eager | argmax XLA |
|---|---|---|
| `[NaN, 2.0, 3.0]` | 2 | 2 | OK (NaN before max) |
| `[1.0, NaN, 3.0]` | 2 | 2 | OK (NaN before max) |
| `[1.0, 2.0, NaN]` | 1 | **2** | BUG (NaN at end) |

## Practical Impact

This bug is triggered by `argmax(softmax(x))` when any element is NaN:

```python
x = tf.constant([[1.0, float('nan'), 2.0]])
print(tf.nn.softmax(x).numpy())
# [[nan, nan, nan]]  — softmax propagates NaN to all outputs

def f(x): return tf.math.argmax(tf.nn.softmax(x), axis=-1)
print(f(x).numpy())             # [0]  — eager: correct
print(tf.function(f, jit_compile=True)(x).numpy())  # [2]  — XLA: wrong
```

NaN commonly appears in ML inputs (missing data, gradient explosion, division by zero). Non-power-of-2 sizes are the norm for vocabulary dimensions, feature counts, class counts, etc. The combination makes this easy to trigger in practice.

## Root Cause

XLA's `argmax`/`argmin` uses a tree reduction. When the array size is not a power of 2, the reduction tree has "leftover" elements. The comparison function used in the reduction returns `false` when either operand is NaN (per IEEE 754: `NaN > x = false`, `NaN < x = false`). In the tree reduction's handling of leftover elements, this causes the later element to always "win" regardless of value, propagating to the final result.

For power-of-2 sizes, the tree is perfectly balanced and every element is compared an equal number of times, masking the NaN comparison issue.

## Environment

- TensorFlow 2.22.0-dev20260429
- Affects both CPU and GPU XLA
- Eager mode: correct
- Grappler (`jit_compile=False`): correct

## Recommendation

File in `openxla/xla`. The XLA `argmax`/`argmin` reduction should handle NaN consistently regardless of array size — either by treating NaN as less than all values (matching eager behavior) or by propagating NaN to the output index, but not by returning an arbitrary last index.

### Standalone code to reproduce the issue

```shell
## Minimal Reproduction


import tensorflow as tf

# Size 3 (non-power-of-2): BUG
x = tf.constant([1.0, 2.0, float('nan')])
print(tf.math.argmax(x).numpy())
# 1  — eager: correct (2.0 is max, at index 1)

print(tf.function(tf.math.argmax, jit_compile=True)(x).numpy())
# 2  — XLA: WRONG (returns last index)

# Size 4 (power-of-2): OK
x4 = tf.constant([1.0, 2.0, float('nan'), 0.5])
print(tf.function(tf.math.argmax, jit_compile=True)(x4).numpy())
# 1  — XLA: correct


Observed on TensorFlow 2.22.0-dev20260429.
```

### Relevant log output

```shell

```

**Comment by dhantule** [comment]
Hi @wuyii8941, 

Thanks for reporting this and sharing the details.
We’ve received the issue and are taking a look. We’ll follow up here once we have more information or if we need any additional inputs.
Appreciate your patience.
